### PR TITLE
Remove psutil usage from regress tests

### PR DESCRIPTION
**Pull request by beeender**
psutil is not a dependency of plpython, thus we cannot assume it exists
when running regress test. We are going to use different python versions
for management utilities and plpython in the future release, and psutil
won't exist anymore while running regress tests.

- Rewrite teh motion_socket tests with 'lsof', which is a explicit
  dependency of greenplum.

Related with #15703

### CODE REVIEW

**Comment by beeender** [comment]
https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-cmulong-15703

testing pipeline

File: src/test/regress/sql/motion_socket.sql [hunk 1]
```diff
@@ -46,18 +48,28 @@ pids_to_check = [r['pid'] for r in res]
...
+        capture_output=True, check=True).stdout
+    plpy.info(
+        f'Checking postgres backend {pid}, ' \
+        f'lsof output:\n{os.linesep.join(map(str, lsof_ret.splitlines()))}')
+    # The first row are the column names
+    r = lsof_ret.splitlines()[1:]
+    for line in r:
+        # No '->' follows he port. The connection has not be setup yet
+        match=re.match(f".*{expected_socket_kind} {hostip}:\d+ .*", str(line))
+        if match != None:
```
Comment (yihong0618) [inline src/test/regress/sql/motion_socket.sql:67]: if not match:

Comment (beeender) [reply src/test/regress/sql/motion_socket.sql]: Changed to `is not None`
refer to https://stackoverflow.com/questions/2710940/python-if-x-is-not-none-or-if-not-x-is-none

Comment (beeender) [reply src/test/regress/sql/motion_socket.sql]: And found a bug :)

**Review by yihong0618 — APPROVED** [review]
LGTM

**Review by higuoxing — APPROVED** [review]
The change looks good to me. But the test looks a little bit complicated. Can we just emit some debug notices when creating motions/sockets and check the output so that we can get rid of PL/Python here?

**Comment by beeender** [comment]
> The change looks good to me. But the test looks a little bit complicated. Can we just emit some debug notices when creating motions/sockets and check the output so that we can get rid of PL/Python here?

That sounds a better plan :P

**Pull request merged**